### PR TITLE
Add procedure to release resources before and after sanity checks are run

### PR DESCRIPTION
**Pull request by pratid**
#### Reviewers

@flopezag, @jframos 
#### Description

Solves issue 4872
#### Testing

Tested on different nodes.

### CODE REVIEW

File: fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py [hunk 1]
```diff
@@ -48,83 +50,110 @@ def __deploy_instance_helper__(self, instance_name, network_name=None, network_c
...
+            if keypair_name:
+                self.nova_operations.create_keypair(keypair_name)
+                self.test_world['keypair_names'].append(keypair_name)
+
+            security_group_name_list = None
+            if sec_group_name:
+                sec_group_id = self.nova_operations.create_security_group_and_rules(sec_group_name)
+                self.test_world['sec_groups'].append(sec_group_id)
+                security_group_name_list = [sec_group_name]
+        except (NovaClientException, NeutronClientException) as e:
```
Comment (flopezag) [inline fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py]: I suggest to have each try-except in order to identify exactly the problem when we see the message.

File: fiware-region-sanity-tests/tests/fiware_region_without_networks_tests.py [hunk 2]
```diff
@@ -43,61 +46,88 @@ def __deploy_instance_helper__(self, instance_name, keypair_name=None, sec_group
...
+            if keypair_name:
+                self.nova_operations.create_keypair(keypair_name)
+                self.test_world['keypair_names'].append(keypair_name)
+
+            security_group_name_list = None
+            if sec_group_name:
+                sec_group_id = self.nova_operations.create_security_group_and_rules(sec_group_name)
+                self.test_world['sec_groups'].append(sec_group_id)
+                security_group_name_list = [sec_group_name]
+        except ClientException as e:
```
Comment (flopezag) [inline fiware-region-sanity-tests/tests/fiware_region_without_networks_tests.py:70]: Idem, I would like separate try-except loop in order to identify clearly when I have a problem with the keypair and when I have a problem with security groups.

File: fiware-region-sanity-tests/tests/fiware_region_base_tests.py [hunk 3]
```diff
@@ -25,146 +25,137 @@
...
+        # skip if not all security groups were deleted
+        if self.test_world['sec_groups']:
+            self.skipTest("Not all the security groups were deleted")
+
+        sec_group_name = TEST_SEC_GROUP_PREFIX
+        try:
+            sec_group_id = self.nova_operations.create_security_group_and_rules(sec_group_name)
+            self.assertIsNotNone(sec_group_id, "Problems creating security group '%s'" % sec_group_name)
+            self.test_world['sec_groups'].append(sec_group_id)
+        except Forbidden as e:
```
Comment (flopezag) [inline fiware-region-sanity-tests/tests/fiware_region_base_tests.py]: Are you sure that the exception forbidden is related to the message "Quota exceeded when creating a ..."?

**Comment by flopezag** [comment]
LGTM

**Comment by jframos** [comment]
Waiting for issues discussed today by Skype...

File: fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py (hunk 1 above)
Comment (pratid) [inline fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py]: Done.

File: fiware-region-sanity-tests/tests/fiware_region_without_networks_tests.py (hunk 2 above)
Comment (pratid) [inline fiware-region-sanity-tests/tests/fiware_region_without_networks_tests.py:70]: Done.

**Comment by jframos** [comment]
LGTM

**Pull request merged**